content/en/blog/_posts/2023-07-18-swap-beta1-graduation/index.md

```diff
@@ -0,0 +1,233 @@
+
+## How is the swap limit being determined with LimitedSwap?
+The configuration of swap memory, including its limitations, presents a significant
+challenge. Not only is it prone to misconfiguration, but as a system-level property, any
+misconfiguration could potentially compromise the entire node rather than just a specific
+workload. To mitigate this risk and ensure the health of the node, we have implemented
+Swap in Beta with automatic configuration of limitations.
+
+With `LimitedSwap`, Pods that do not fall under the Burstable QoS classification (i.e.
+`BestEffort`/`Guaranteed` Qos Pods) are prohibited from utilizing swap memory.
+`BestEffort` QoS Pods exhibit unpredictable memory consumption patterns and lack
+information regarding their memory usage, making it difficult to determine a safe
+allocation of swap memory. Conversely, `Guaranteed` QoS Pods are typically employed for
+applications that rely on the precise allocation of resources specified by the workload,
+with memory being immediately available. To maintain the aforementioned security and node
+health guarantees, these Pods are not permitted to use swap memory when `LimitedSwap` is
+in effect.
+
+Prior to detailing the calculation of the swap limit, it is necessary to define the following terms:
+* `nodeTotalMemory`: The total amount of physical memory available on the node.
+* `totalPodsSwapAvailable`: The total amount of swap memory on the node that is available for use by Pods (some swap memory may be reserved for system use).
+* `containerMemoryRequest`: The container's memory request.
+
+Swap limitation is configured as:
+`(containerMemoryRequest / nodeTotalMemory) * totalPodsSwapAvailable`.
+
+In other words, the amount of swap that a container is able to use is proportionate to its
+memory request, the node's total physical memory and the total amount of swap memory on
+the node that is available for use by Pods.
+
+It is important to note that, for containers within Burstable QoS Pods, it is possible to
+opt-out of swap usage by specifying memory requests that are equal to memory limits.
+Containers configured in this manner will not have access to swap memory.
+
+## How does it work?
+There are a number of possible ways that one could envision swap use on a node.
+When swap is already provisioned and available on a node,
+[we have proposed](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#proposal)
+the kubelet should be able to be configured such that:
+- It can start with swap on.
+- It will direct the Container Runtime Interface to allocate zero swap memory
+  to Kubernetes workloads by default.
+- You can configure the kubelet to specify swap utilization for the entire
+  node.
+
+Swap configuration on a node is exposed to a cluster admin via the
+[`memorySwap` in the KubeletConfiguration](/docs/reference/config-api/kubelet-config.v1beta1/).
```

Comment:
404 page not found: https://github.com/kubernetes/website/blob/b0de0bbccc63b37fb3c2c67a15db44fe8fbb261b/docs/reference/config-api/kubelet-config.v1beta1

Reply:
This part was donated by @pacoxu. Which link did you wish to use here? Can you paste the external link and then we can try to understand how to use the internal one?

Reply:
https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/

The markdown in GitHub may be missing leading to a wrong URL.  This should be OK, in my opinion.

Reply:
link to this should work: /content/en/docs/reference/config-api/kubelet-config.v1.md

Reply:
> The markdown in GitHub may be missing leading to a wrong URL. This should be OK, in my opinion.

Ack.

Reply:
The link in preview website: https://deploy-preview-42009--kubernetes-io-main-staging.netlify.app/blog/2023/07/18/swap-beta1-1.28-2023/#how-does-it-work can work well.

Reply:
Fixed all broken links, with only internal references :+1: 
Thsnks a lot @pacoxu and @chendave!